internal/resource/rabbitmq_resource_builder.go

```diff
@@ -22,6 +22,7 @@ type RabbitmqResourceBuilder struct {
 type ResourceBuilder interface {
 	Update(runtime.Object) error
 	Build() (runtime.Object, error)
+	RequiresClusterRestart() bool
```

Comment:
Small: I find the name of this method a bit confusing. Without the context, the name suggests that it should be the single method needed to be called to determine whether to restart the cluster or not. However, when using the method, we still need to check the operational result.

Maybe rename it to "UpdateRequiresStsRestart"? so that it's a bit more specific? Your call.